### PR TITLE
R4R : Bug fix and add test case

### LINKED ISSUE #575: Sequencer: compareSequencerSet test and fix
In the current implementation,`compareSequencerSet` cannot effectively identify the deleted sequencer. This bug needs to be fixed and the test cases of this method should be supplemented.

### PR DESCRIPTION
**Pull request by Kukoomomo**
close #575

### CODE REVIEW

**Review by curryxbo — APPROVED** [review]

**Review by wukongcheng — APPROVED** [review]

File: l2geth/consensus/clique/scheduler.go [hunk 1]
```diff
@@ -385,21 +385,42 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
 
 // compareSequencerSet will return the update with Driver.seqz
 func compareSequencerSet(old []*Sequencer, newSeq synchronizer.SequencerSequencerInfos) []*Sequencer {
+	var find []common.Address
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:388]: How about using mapping here?

File: l2geth/consensus/clique/scheduler.go [hunk 2]
```diff
@@ -384,22 +384,47 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
 }
 
 // compareSequencerSet will return the update with Driver.seqz
-func compareSequencerSet(old []*Sequencer, newSeq synchronizer.SequencerSequencerInfos) []*Sequencer {
+func compareSequencerSet(preSeqs []*Sequencer, newSeq synchronizer.SequencerSequencerInfos) []*Sequencer {
+	var notDel map[common.Address]bool
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:388]: init the map

File: l2geth/consensus/clique/scheduler.go [hunk 3]
```diff
@@ -384,22 +384,47 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
...
+	// select the deleted sequencer
+	for _, v := range preSeqs {
+		del := true
+		for addr, _ := range notDel {
+			// find in new sequencer set, this sequencer still in sequencer set
+			if bytes.Equal(addr.Bytes(), v.Address.Bytes()) {
+				del = false
+				break
+			}
+		}
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:420]: It is not necessary to iterate the map. Just read the value with the address as the key.

Comment (Kukoomomo) [reply l2geth/consensus/clique/scheduler.go:420]: LGTM

File: l2geth/consensus/clique/scheduler.go [hunk 4]
```diff
@@ -384,22 +384,44 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
...
-				changed = false
-				break
+			// found the sequencer and their power equal
+			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) {
+				// find the sequencer in previous sequencer set
+				notDel[seq.Address] = true
+				if power.Int64() == seq.Power {
+					// voting power have not change
+					changed = false
+					break
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:401]: Once the address is matched, no matter whether the power is equal or not, the inner loop should be broken.

File: l2geth/consensus/clique/scheduler.go [hunk 5]
```diff
@@ -384,22 +384,44 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
...
 			power := big.NewInt(1).Div(v.Amount, scale)
-			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) && power.Int64() == seq.Power {
-				changed = false
-				break
+			// found the sequencer and their power equal
+			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) {
+				// find the sequencer in previous sequencer set
+				notDel[seq.Address] = true
+				if power.Int64() == seq.Power {
+					// voting power have not change
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:399]: `// voting power have not change` -> `// voting power didn't change`

File: l2geth/consensus/clique/scheduler.go [hunk 6]
```diff
@@ -384,22 +384,44 @@ func (schedulerInst *Scheduler) syncSequencerSetRoutine() {
...
 	var tmp synchronizer.SequencerSequencerInfos
 	for i, v := range newSeq {
 		changed := true
-		for _, seq := range old {
+		for _, seq := range preSeqs {
 			power := big.NewInt(1).Div(v.Amount, scale)
-			if bytes.Equal(seq.Address.Bytes(), v.MintAddress.Bytes()) && power.Int64() == seq.Power {
-				changed = false
-				break
+			// found the sequencer and their power equal
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/scheduler.go:394]: `// found the sequencer and their power equal` -> `// figure out unchanged sequencer`

File: l2geth/consensus/clique/scheduler.go (hunk 4 above)
Comment (Kukoomomo) [reply l2geth/consensus/clique/scheduler.go:401]: LGTM

**Review by curryxbo — APPROVED** [review]

**Review by HaoyangLiu — APPROVED** [review]

**Pull request merged**